src/app/participant/check-in/page.tsx

```diff
@@ -1,14 +1,27 @@
 "use client";
 
-// THIS PAGE IS A SAMPLE, PLEASE REPLACE IT AND IMPROVE IT FROM HERE
-// app/food/page.tsx
 import { generateClient } from "aws-amplify/api";
 import { getCurrentUser } from "aws-amplify/auth";
+import Image from "next/image";
+import Link from "next/link";
+import { useEffect, useState } from "react";
 
 import { type Schema } from "@/amplify/data/resource";
 
-export default function FoodPage() {
+const check_mark_icon = "/svgs/checkin/check_mark.svg";
+const cross_icon = "/svgs/checkin/circle_cross.svg";
+
+const CHECKIN_STATUS_TILE_STLYES =
+  "my-20 flex w-4/5 max-w-[1000px] flex-col items-center rounded-xl border-2 border-dark-pink bg-white p-10 shadow-[15px_15px_0px_0px_#FF4D6F]";
+const CHECKIN_STATUS_HEADER_STYLES = "mb-2 text-2xl font-bold text-dark-pink";
+const CHECKIN_STATUS_TEXT_STYLES =
+  "mb-6 max-w-[450px] text-center text-lg text-black";
+const CHECKIN_STATUS_BUTTON_STYLES =
+  "rounded-xl bg-dark-pink p-4 font-bold hover:bg-pastel-pink";
+
+const CheckInPage = () => {
   const client = generateClient<Schema>();
+  const [status, setStatus] = useState("loading");
```

Comment:
Can you make the loading, error, checked in states an ENUM and use those. Strings are not type safe